### PR TITLE
add redux saga support and wire up login

### LINKED ISSUE #94: Add redux-saga to web project
Add `redux` + `redux-saga`  to web UI. 

## Expected Behaviour

- `redux` for storing app state, e.g. product data, user tokens.
- `redux-saga` for side-effects, e.g. fetching product data from `data-service`, authenticating user with `auth-service`.

## Current Behaviour

N/A.

## Possible Solution

- https://redux.js.org/.
- https://redux-saga.js.org (see also [redux "thunks"](https://github.com/reduxjs/redux-thunk) ).

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #94 

## Description
Adds redux saga support and creates saga for user login
Also added .env entries - the `.local.env` is excluded from git and is useful for configuring your local environment.

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [ ] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [ ] Tests OK.

### CODE REVIEW

**Comment by mark-prins** [comment]
draft until I can get the problem with CORS sorted out!

**Comment by mark-prins** [comment]
I'm thinking to give up on CORS ( for now ) at least to get this PR in. Then we can work on redux+apollo and get the other parts tidy, and finish off the auth in another PR.
Will just fix up those conflicts now.

File: nx-workspace/package.json [hunk 1]
```diff
@@ -49,6 +49,7 @@
     "react": "16.13.1",
     "react-dom": "16.13.1",
     "react-router-dom": "^5.2.0",
+    "redux-saga": "^1.1.3",
```
Comment (katherine-sussol) [inline nx-workspace/package.json:52]: Hm - is it just me getting a compilation error about react-redux missing? (in particular, `web/src/components/templates/Login.tsx` is looking for it)
Looking at the history it looks like some redux things got removed when master was merged in earlier today?

Comment (mark-prins) [reply nx-workspace/package.json:53]: good catch, thanks! I have accidentally installed it globally and caught myself out.
Added now. a8f545118

**Pull request merged**